javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/bytebuddy/matcher/ClassLoaderMatcher.java

```diff
@@ -43,6 +44,7 @@ private ClassLoaderHasClassesNamedMatcher(String... classNames) {
       for (int i = 0; i < resources.length; i++) {
         resources[i] = resources[i].replace(".", "/") + ".class";
       }
+      ClassLoaderMatcherCacheHolder.addCache(cache);
```

Comment:
Any thoughts on storing `URLClassLoader.getUrls().length` into the cache instead of `Boolean` and invalidating here if it doesn't match? The nice property of URLClassPath is it can only be added to, not removed from.

Pros
- Less instrumentation
- Less bootstrap classpath logic
- Doesn't split logic in two places

Cons
- Slower when matching, `URL[]` copy when calling `getURLs`. But benchmarks may not find it to be a big deal. 
  - If only URLClassLoader had a `size` method :(

Reply:
oh, that's a good idea, bummer about the `URL[]` copy, e.g. war file with 100 jars would have to make a 100-element copy on every `hasClassesNamed` match (which looks like there are ~100, and I think they all have to trigger for each class load)